src/troute-network/troute/nhd_io.py

```diff
@@ -2128,9 +2142,86 @@ def write_flowveldepth_netcdf(stream_output_directory, file_name,
                 'code_version': '',
             }
         )
+def stream_output_mask_reader(stream_output_mask):
+    if not stream_output_mask:
+        return {}
+    with open(stream_output_mask, 'r') as file:
+        mask_list = yaml.safe_load(file)
+    
+    return mask_list
+
+def mask_find_seg(mask_list, nexus_dict, poi_crosswalk):
+    seg_id = []
+    nex_id = {}
+    
+    if not mask_list:
+        return nex_id, seg_id
+    if 'wb' in mask_list and mask_list['wb']:
+        if 9999 not in mask_list['wb']: 
+            seg_id.extend(mask_list['wb'])
+        else:
+            seg_id.extend([9999])
+
+    if 'nex' in mask_list and mask_list['nex']:
+        if 9999 in mask_list['nex']:
+            
+            for key, val in nexus_dict.items():
+                nex_key = int(key.split('-')[-1])
+                nex_id[nex_key] = [int(v.split('-')[-1]) for v in val]
+        else:
+            for key in mask_list['nex']:
+                item = f'nex-{key}'
+                nex_id[key] = [int(val.split('-')[-1]) for val in nexus_dict.get(item, [])]
+
+    return nex_id, seg_id
+
+def updated_flowveldepth(flowveldepth, nex_id, seg_id, mask_list):
+    flowveldepth.index.name = 'featureID'
+    flowveldepth['Type'] = 'wb'
+    flowveldepth.set_index('Type', append=True, inplace=True)
+
+    filtered_v_columns = [col for col in flowveldepth.columns if col[1] == 'v']
+    filtered_d_columns = [col for col in flowveldepth.columns if col[1] == 'd']
+
+    def create_mask(ids):
+        if 9999 in ids:
+            ids = flowveldepth.index.get_level_values('featureID')
+        
+        masked = (flowveldepth.index.get_level_values('featureID').isin(ids)) & (flowveldepth.index.get_level_values('Type') == 'wb')
+        return masked
+    
+    flowveldepth_seg = flowveldepth[create_mask(seg_id)] if seg_id else pd.DataFrame()
+
+    nex_data_list = []
+    if nex_id:
+        for nex_key, nex_val in nex_id.items():
+            new_index = pd.MultiIndex.from_tuples([(nex_key, 'nex')], names=['featureID', 'Type'])
+            flowveldepth_nex = flowveldepth[create_mask(nex_val)]
+            
+            if not flowveldepth_nex.empty:
+                if len(flowveldepth_nex.index) > 1:
+                    summed_data = flowveldepth_nex.sum()
+                    new_data = pd.DataFrame([summed_data], index=new_index)
+                    new_data.loc[(nex_key, 'nex'), filtered_v_columns] = np.nan
+                    new_data.loc[(nex_key, 'nex'), filtered_d_columns] = flowveldepth_nex[filtered_d_columns].mean()
+                else:
+                    new_data = flowveldepth_nex.copy()
+                    new_data.index = new_index
+
+                nex_data_list.append(new_data)
+        
+        all_nex_data = pd.concat(nex_data_list) if nex_data_list else pd.DataFrame()
+    else:
+        all_nex_data = pd.DataFrame()
```

Comment:
This chunk seems quite slow if 9999 is included. I wonder if this could be sped up by adding a column to `flowveldepth` that is the `toid` for each `wb-` `featureID`. These would obviously be `nex-` IDs, then you could just group by the `toid` column and sum? Something like that.